go-controller/pkg/node/management-port.go

```diff
@@ -23,16 +32,42 @@ func createManagementPort(nodeNameOriginal string, hostSubnets []*net.IPNet, nod
 	// will try to fetch and what kubernetes provides, thus failing to
 	// create the port on the logical switch.
 	// Until the above is changed, switch to a lowercase hostname
-	nodeName := strings.ToLower(nodeNameOriginal)
+	nodeName = strings.ToLower(nodeName)
 
+	switch config.OvnKubeNode.Mode {
+	default:
+		return &managementPort{
+			nodeName:    nodeName,
+			hostSubnets: hostSubnets,
+		}
+	case types.NodeModeSmartNIC:
+		return &managementPortSmartNIC{
+			nodeName:    nodeName,
+			hostSubnets: hostSubnets,
+			vfRepName:   config.OvnKubeNode.MgmtPortNetdev,
```

Comment:
We assume vf rep has the same name as vf?

Reply:
we do not assume it.

when ovnkube node is deployed on DPU (ovnkube-node-mode == smart-nic) the value needs to point to the VF representor matching the netdev associated with the management port on the host.

An improvement to this would be to derive one from the other.
that is, similar to ovnkube-node in smart-nic-host mode setting the Pod's connection-details in pod spec.
ovnkube-node in smart-nic-host mode would set mgmt-connection-details in the node annotation. then ovnkube-node in smart-nic mode running on the DPU would set up the management port (find representor and configure it)